app/src/main/java/org/kiwix/kiwixmobile/main/MainActivity.java

```diff
@@ -587,6 +585,23 @@ void openToc() {
     drawerLayout.openDrawer(GravityCompat.END);
   }
 
+  @Override public void onBackPressed() {
+    if (tabSwitcherRoot.getVisibility() == View.VISIBLE) {
+      selectTab(currentWebViewIndex);
+      hideTabSwitcher();
+    } else if (getCurrentWebView().canGoBack()) {
+      getCurrentWebView().goBack();
+    } else if (isFullscreenOpened) {
+      closeFullScreen();
+    } else if (compatCallback.mIsActive) {
+      compatCallback.finish();
+    } else if (drawerLayout.isDrawerOpen(GravityCompat.END)) {
```

Comment:
This should be moved above `getCurrentWebView().canGoBack()` otherwise, the web view will go to the previous page instead of the drawer getting closed.

Reply:
good catch, I was so annoyed at the overriding `onKeyDown` I didn't pay enough attention